sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,593 @@
+import { RuleResourceSerializer, RuleOptions } from "./serializers/ruleResourceSerializer";
+
+import {
+  ListRequestOptions,
+  QueueResult,
+  ListQueuesResult,
+  TopicResult,
+  ListTopicsResult,
+  SubscriptionResult,
+  ListSubscriptionsResult,
+  RuleResult,
+  ListRulesResult
+} from "./managementOperationInterfaces";
+
+/**
+ * All internal operations return a `Promise<HttpOperationResponse>` object of which the `parsedBody`
+ * is populated with the response in JSON format with following structure:
+ *  {
+ *   response:
+ *      {
+ *        isSuccessful: isSuccessful,
+ *        statusCode: statusCode,
+ *        body: body,
+ *        headers: headers
+ *      },
+ *   result: <parsed result, if any>,
+ *   error: <error information, if any>
+ *  }
+ *
+ * Above information is used to construct and return a more strongly typed resource as `QueueResult`,
+ * `TopicResult`, `SubscriptionResult` or `RuleResult`.
+ *
+ */
+export class ServiceBusAtomManagementClient extends ServiceClient {
+  private endpoint: string;
+
+  private queueResourceSerializer: AtomXmlSerializer;
+  private topicResourceSerializer: AtomXmlSerializer;
+  private subscriptionResourceSerializer: AtomXmlSerializer;
+  private ruleResourceSerializer: AtomXmlSerializer;
+
+  /**
+   * Initializes a new instance of the ServiceBusManagementClient class.
+   * @param credentials Credentials needed for the client to connect to Azure.
+   * @param options The parameter options
+   */
+  constructor(connectionString: string, proxySettings?: ProxySettings) {
+    const connectionStringObj = ServiceBusAtomManagementClient.parseConnectionString(
+      connectionString
+    );
+    const credentials = new SasServiceClientCredentials(
+      connectionStringObj.SharedAccessKeyName,
+      connectionStringObj.SharedAccessKey
+    );
+
+    const requestPolicyFactories: RequestPolicyFactory[] = [];
+    requestPolicyFactories.push(userAgentPolicy());
+    requestPolicyFactories.push(logPolicy(httpAtomXml));
+    requestPolicyFactories.push(atomSerializationPolicy());
+    requestPolicyFactories.push(signingPolicy(credentials));
+
+    if (proxySettings) {
+      requestPolicyFactories.push(proxyPolicy(proxySettings));
+    }
+    const serviceClientOptions: ServiceClientOptions = {
+      requestPolicyFactories: requestPolicyFactories
+    };
+
+    super(credentials, serviceClientOptions);
+    this.queueResourceSerializer = new QueueResourceSerializer();
+    this.topicResourceSerializer = new TopicResourceSerializer();
+    this.subscriptionResourceSerializer = new SubscriptionResourceSerializer();
+    this.ruleResourceSerializer = new RuleResourceSerializer();
+    this.endpoint = (connectionString.match("Endpoint=sb://(.*)/;") || "")[1];
+  }
+
+  /**
+   * Creates a queue.
+   * @param queuePath Name of the queue to use
+   * @param queueOptions
+   */
+  async createQueue(queuePath: string, queueOptions: QueueOptions): Promise<QueueResult> {
```

Comment:
Consider the below renames
queuePath -> name
queueOptions -> options

Since the name of the method `createQueue` already sets the context that we are dealing with queues, the names of the input arguments can be simplified.

Also, use `name` instead of `path` because from user perspective they are giving us a name and they are not aware of any "path" which is a service side concept after the entity is created

Reply:
Also, shouldn't `queueOptions` be optional? Same for options in all other cases except for update operations and for creating rule

Reply:
Updated to use `name` in all applicable places.
Also updated the create and update APIs to reflect the optional options as applicable